src/server.rs

```diff
@@ -222,14 +224,27 @@ impl Server {
         let event_loop = (*event_loop).call_method0("run_forever");
         if event_loop.is_err() {
             debug!("Ctrl c handler");
-            Python::with_gil(|py| {
-                pyo3_asyncio::tokio::run(py, async move {
-                    execute_event_handler(shutdown_handler, &task_locals.clone())
-                        .await
-                        .unwrap();
-                    Ok(())
-                })
-            })?;
+
+            if let Some(function) = shutdown_handler {
+                if function.is_async {
+                    debug!("Shutdown event handler async");
+
+                    pyo3_asyncio::tokio::run_until_complete(
+                        task_locals.event_loop(py),
+                        pyo3_asyncio::into_future_with_locals(
+                            &task_locals.clone(),
+                            function.handler.as_ref(py).call0()?,
+                        )
+                        .unwrap(),
+                    )
+                    .unwrap();
```

Comment:
this should be a separate function. called `execute_shutdown_handler` in the same file as https://github.com/sparckles/Robyn/pull/965/files#diff-e57d319b832ca59f61da275f35a0903e6a4f6a0fbfee4d2cfb149beb1f7aa220

Reply:
doing this from the same file (i.e, without starting an async function) was the solution to our issue.

now, if I try to create a function & move the code there, I get `unused implementer of `futures_util::Future` that must be used futures do nothing unless you `.await` or poll them`. now, if I add `await`, I get the error ``await` is used inside non-async function, which is not an `async` context`. which can be solved by starting a new async function & we are back to square one.

or, am I missing something? do you have any other suggestions for me to try? 🤔

Reply:
@VishnuSanal , I don't see you awaiting anything in the code. What do you mean by `.await` or polling it here?

Reply:
> doing this from the same file (i.e, without starting an async function) was the solution to our issue.

now, if I try to create a function & move the code there, I get unused implementer of futures_util::Futurethat must be used futures do nothing unless you.await or poll them. now, if I add await, I get the error ``awaitis used inside non-async function, which is not anasync` context`. which can be solved by starting a new async function & we are back to square one.

or, am I missing something? do you have any other suggestions for me to try? 🤔


also mention this in the code